official/utils/logs/logger.py

```diff
@@ -113,21 +114,56 @@ def log_run_info(self, model_name):
     run_info = {
         "model_name": model_name,
         "machine_config": {},
-        "run_date": datetime.datetime.now().strftime(_DATE_TIME_FORMAT_PATTERN)}
+        "run_date": _formatted_date(self.run_date)
+    }
+
     _collect_tensorflow_info(run_info)
     _collect_tensorflow_environment_variables(run_info)
     _collect_cpu_info(run_info)
     _collect_gpu_info(run_info)
     _collect_memory_info(run_info)
 
+    self._log_run_info(run_info)
+
+  def log_run_end(self):
+    """Log timing information at the end of a run."""
+    end_date = datetime.datetime.now()
+    delta = end_date - self.run_date
+    end_info = {
+        "end_date": _formatted_date(end_date),
+        "elapsed_time": str(delta),
+        "elapsed_seconds": delta.total_seconds()
```

Comment:
Since those two field are just derived data, can we generate them on the repository side?